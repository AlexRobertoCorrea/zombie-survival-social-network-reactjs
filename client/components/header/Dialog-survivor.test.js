import React from 'react';

import DialogSurvivor from './Dialog-survivor';

describe('<DialogSurvivor />', () => {
  const open = true;
  const handleCloseDialogSurvivor = () => {};
  const handleSurvivorPerson = () => {};
  const options = {
    form: {},
    handleCloseDialogSurvivor,
    notifier: {},
    store: {
      peopleStore: {
        fetchingData: false,
        people: [],
        person: {},
        savingData: false
      }
    }
  };
  const fields = {
    name: {},
    age: {},
    gender: {},
    latitude: {},
    longitude: {},
    water: {},
    food: {},
    medication: {},
    ammunition: {}
  };
  
  const dialogSurvivor = shallow(<DialogSurvivor
    open={open}
    handleCloseDialogSurvivor={handleCloseDialogSurvivor}
    handleSurvivorPerson={handleSurvivorPerson}
    options={options}
    fields={fields}
  />);
  
  it('renders the dialogSurvivor as expected', () => {
    expect(dialogSurvivor).toMatchSnapshot();
  });
});
