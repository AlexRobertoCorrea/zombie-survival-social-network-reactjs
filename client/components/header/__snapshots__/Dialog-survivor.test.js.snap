// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogSurvivor /> renders the dialogSurvivor as expected 1`] = `
<SimpleDialog
  options={
    Object {
      "content": <FormPerson
        fields={
          Object {
            "age": Object {},
            "ammunition": Object {},
            "food": Object {},
            "gender": Object {},
            "latitude": Object {},
            "longitude": Object {},
            "medication": Object {},
            "name": Object {},
            "water": Object {},
          }
        }
        form={Object {}}
        gendersOptions={
          Array [
            Object {
              "label": "Male",
              "value": "M",
            },
            Object {
              "label": "Female",
              "value": "F",
            },
          ]
        }
        handleSurvivorPerson={[Function]}
        options={
          Object {
            "form": Object {},
            "handleCloseDialogSurvivor": [Function],
            "notifier": Object {},
            "store": Object {
              "peopleStore": Object {
                "fetchingData": false,
                "people": Array [],
                "person": Object {},
                "savingData": false,
              },
            },
          }
        }
      />,
      "leftButtonLabel": "Cancel",
      "leftButtonOnClick": [Function],
      "onClose": [Function],
      "open": true,
      "rightButtonLabel": "Save",
      "rightButtonOnClick": [Function],
      "title": "Register user",
    }
  }
/>
`;
